git-make-release

```diff
@@ -2,19 +2,37 @@
 
 EDITOR=${EDITOR:-editor}
 RELEASE_BRANCH=0
+RELEASE_PATCH=0
+RELEASE_MINOR=0
+RELEASE_MAJOR=0
 
 # Parse options
-OPTS=$(getopt -o b --long release-branch -n 'parse-options' -- "$@")
+OPTS=$(getopt -o b,p,m,M --long release-branch,patch,minor,major -n 'parse-options' -- "$@")
 eval set -- "$OPTS"
 
 while true; do
     case "$1" in
         -b | --release-branch ) RELEASE_BRANCH=1 ; shift ;;
+        -p | --patch ) RELEASE_PATCH=1 ; shift ;;
+        -m | --minor ) RELEASE_MINOR=1 ; shift ;;
+        -M | --major ) RELEASE_MAJOR=1 ; shift ;;
         -- ) shift; break ;;
         * ) break ;;
     esac
 done
 
+# RELEASE_PATCH, RELEASE_MINOR and RELEASE_MAJOR are mutually exclusive
+if [ $((RELEASE_PATCH + RELEASE_MINOR + RELEASE_MAJOR)) -gt 1 ]; then
+  echo "Options --patch, --minor and --major are mutually exclusive!"
+  exit 1
+fi
+
+# patch, minor and major are not allowed in combination with release-branch
+if [ $((RELEASE_PATCH + RELEASE_MINOR + RELEASE_MAJOR)) -gt 0 -a $RELEASE_BRANCH -eq 1 ]; then
+  echo "Options --patch, --minor and --major are not allowed in combination with --release-branch!"
+  exit 1
+fi
+
```

Comment:
Isn't making an occasional PATCH release the whole point of having a branch for release?

Reply:
true, but dealing with branch releases would be rather different, since we would need to make the PR/MR into that branch - or merge directly if we assume non-default branches are never protected?